dbms/src/TestUtils/FunctionTestUtils.cpp

```diff
@@ -103,22 +113,117 @@ ::testing::AssertionResult columnEqual(
     return columnEqual(expected.column, actual.column);
 }
 
-void blockEqual(
+::testing::AssertionResult blockEqual(
     const Block & expected,
     const Block & actual)
 {
     size_t columns = actual.columns();
     size_t expected_columns = expected.columns();
 
-    ASSERT_EQ(expected_columns, columns);
+    ASSERT_EQUAL(expected_columns, columns, "Block size mismatch");
 
     for (size_t i = 0; i < columns; ++i)
     {
         const auto & expected_col = expected.getByPosition(i);
         const auto & actual_col = actual.getByPosition(i);
-        ASSERT_EQ(actual_col.type->getName(), expected_col.type->getName());
-        ASSERT_COLUMN_EQ(expected_col.column, actual_col.column);
+        auto cmp_res = columnEqual(expected_col, actual_col);
+        if (!cmp_res)
+            return cmp_res;
     }
+    return ::testing::AssertionSuccess();
+}
+
+/// @pre size of each column should be the same
```

Comment:
pre?What does it mean

Reply:
Oh, it might not be a good idea to add "@pre". I will remove it. It actually means precondition.